src/views/News/index.js

```diff
@@ -43,465 +70,556 @@ const styles = {
-          "to the appropriate 'state' property value " +
-          'for the onChange() function to work',
-      );
+    // If no error occured and the cropper is shown, the cropper text is displayed
+    else if (cropperImageData) {
+      message = 'Crop Photo to liking & Click Submit';
     }
-    this.setState({ [event.target.name]: event.target.value });
+
+    // If no error occured and the cropper is not shown, the pick a file text is displayed
+    else {
+      message = isMobile
+        ? 'Tap Image to Browse Files'
+        : 'Drag & Drop Picture, or Click to Browse Files';
+    }
+    return message;
   }
 
-  callFunction(functionName, param) {
-    if (functionName == null) {
-      throw new Error('Function name not specified to callFunction (news)');
+  // Method called when 'edit' clicked for a news item
+  async function handleNewsItemEdit(newsID) {
+    let newsItem = await newsService.getPostingByID(newsID);
+
+    setOpenPostActivity(true);
+    setNewPostCategory(newsItem.categoryID);
+    setNewPostSubject(newsItem.Subject);
+    setNewPostBody(newsItem.Body);
+    setCurrentlyEditing(newsID);
+
+    /*
+    Error checking. Theoretically, this code is designed so that
+    When the get method in the API service is called from the frontend,
+    it will return the image data, even though that's not the value
+    of the image column in news entries. But in the impossible event that
+    it somhow DID return the path of the image instead of the image data,
+    not only would that produce garbage and make the cropper have trouble,
+    but it also is a potential security concern; it sends data back to the
+    client that shouldn't be sent anywhere.
+    */
+    let base64Test = /^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$/;
+
+    if (base64Test.test(newsItem.Image) && newsItem.Image !== null) {
+      let newImageData = 'data:image/jpg;base64,' + newsItem.Image;
+      setCropperImageData(newImageData);
+    } else {
+      setCropperImageData(null);
     }
-    switch (functionName) {
-      case 'updateSnackbar':
-        if (param == null) {
-          throw new Error("callFunction 'Update Snackbar' requires a parameter (news)");
-        }
-        this.updateSnackbar(param);
-        break;
-      case 'handleNewsItemEdit':
-        if (param == null) {
-          throw new Error("callFunction 'handleNewsItemEdit' requires a parameter (news)");
-        }
-        this.handleNewsItemEdit(param);
-        break;
-      default:
-        console.log('callFunction function name not applicable, double check your parameter');
+  }
+
+  /**
+   * Handles closing the Photo Updater Dialog Box
+   */
+  /*
+  async function handleCloseCancel() {
+    setOpenPhotoDialog(false);
+    setShowCropper(null);
+    clearPhotoDialogErrorTimeout();
+  }
+  */
+
+  function onCropperZoom(event) {
+    if (event.detail.ratio > 1) {
+      event.preventDefault();
+      React.cropperRef.current.cropper.zoomTo(1);
```

Comment:
`cropperRef` is a local variable, not a property on React. Shouldn't this be:
```suggestion
     cropperRef.current.cropper.zoomTo(1);
```